### PR TITLE
mget: handle empty results

### LINKED ISSUE #2954: search/api.py:mget does not handle properly empty search results
https://sentry.inspirehep.net/inspire-sentry/prod/issues/179/

```
KeyError: '_source'
(12 additional frame(s) were not displayed)
...
  File "inspirehep/modules/records/serializers/response.py", line 44, in view
    serializer.serialize(pid, record, links_factory=links_factory),
  File "inspirehep/modules/records/serializers/impactgraph_serializer.py", line 99, in serialize
    'earliest_date'
  File "inspirehep/utils/record_getter.py", line 89, in get_es_records
    return search_class.mget(uuids, **kwargs)
  File "inspirehep/modules/search/api.py", line 93, in mget
    results = [document['_source'] for document in documents['docs']]

KeyError: '_source'
```

### PR DESCRIPTION
**Pull request by david-caro**
- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Fixes #2954

Signed-off-by: David Caro <david@dcaro.es>

### CODE REVIEW

File: inspirehep/modules/search/api.py [hunk 1]
```diff
@@ -90,7 +90,10 @@ def mget(self, uuids, **kwargs):
                 body={'ids': uuids},
                 **kwargs
             )
-            results = [document['_source'] for document in documents['docs']]
+            results = [
+                document['_source'] for document in documents['docs']
+                if document.get('found')
```
Comment (jmartinm) [inline inspirehep/modules/search/api.py]: So when there are no real matches, there is still one document but with no `found` key?

Comment (david-caro) [reply inspirehep/modules/search/api.py]: kind of:
```
{
    u'_id': u'7f6bd778-c2f8-4ddf-a98d-34db4024544c',
    u'_index': u'records-hep',
    u'_type': u'hep',
    u'found': False
}
```

Comment (jmartinm) [reply inspirehep/modules/search/api.py]: Ah I see 👍

**Comment by david-caro** [comment]
You might be right that that error is happening, but it any case I don't think that this function is the one that should be responsible of knowing that, should be the function calling this the one that should know if the uuids it's looking for should or should not exist and throw an error (or handle however needed).

**Comment by jacquerie** [comment]
> You might be right that that error is happening, but it any case I don't think that this function is the one that should be responsible of knowing that, should be the function calling this the one that should know if the uuids it's looking for should or should not exist and throw an error (or handle however needed).

Thing is, we shouldn't need this function at all, because this could just be a `get_records`. I need a diagram to explain the design error that led to the creation of this function... will you be back on Monday?

**Comment by david-caro** [comment]
yep

**Comment by jacquerie** [comment]
This can be closed for the reasons contained in https://github.com/inspirehep/inspire-next/pull/2955#pullrequestreview-78792449.

**Pull request closed without merging**